velox/exec/Window.h

```diff
@@ -22,6 +22,16 @@
 
 namespace facebook::velox::exec {
 
+// Structure for k preceding/following frame arguments, constantValue stores
+// the value of k for constant k and constantIndexValue stores the index of the
+// column when k is a column. When k is a column, columnarOffset stores the
+// offsets for the rows being parsed in the call to 'apply'.
+struct FrameArg {
```

Comment:
This code can be simplified a bit. In this struct you can have column_index_t index and VectorPtr<int64_t> value. The VectorPtr can be a ConstantVector when k is a constant. index is kConstantChannel when k is a constant. This would simplify your code.